chainer/link.py

```diff
@@ -130,15 +133,21 @@ def add_param(self, name, shape, dtype=numpy.float32):
                 name will be removed.
             shape (int or tuple of ints): Shape of the parameter array.
             dtype: Data type of the parameter array.
+            initializer(chainer.initializer.Initializer): If not ``None``,
+                the data is initialized by initializer. In that case,
+                ``dtype`` argument is ignored.
```

Comment:
If it is not ``None``, the data is initialized with the given initializer.
Note that in this case ``dtype`` argument is ignored.